iptables/iptables.go

```diff
@@ -330,12 +357,19 @@ func Exists(table Table, chain string, rule ...string) bool {
 	if supportsCOpt {
 		// if exit status is 0 then return true, the rule exists
 		_, err := Raw(append([]string{"-t", string(table), "-C", chain}, rule...)...)
+		if err != nil {
+			_, err = Raw(append([]string{"-t", string(table), "-C", chain}, append(rule, commentArgs...)...)...)
+		}
 		return err == nil
 	}
 
 	// parse "iptables -S" for the rule (it checks rules in a specific chain
 	// in a specific table and it is very unreliable)
-	return existsRaw(table, chain, rule...)
+	if existsRaw(table, chain, rule...) {
```

Comment:
It's not just comment that gets reordered. How does it work currently?

```
(compute)root@haha:~# iptables -t nat -S
-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
(compute)root@haha:~# iptables -t nat -A PREROUTING -p tcp -d 192.168.1.1 --dport 1234 -j DNAT --to-destination 172.17.0.1:4321 ! -i lo
(compute)root@haha:~# iptables -t nat -S
-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
-A PREROUTING -d 192.168.1.1/32 ! -i lo -p tcp -m tcp --dport 1234 -j DNAT --to-destination 172.17.0.1:4321
```

iptables 1.4.11 came out in 2011 (5 years ago), I wonder if anyone uses it with docker:

* http://www.netfilter.org/news.html#2011-05-26

Quick googling reveals that debian 6.0.3 had linux kernel 2.6.32.46 at this time.